clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -369,6 +390,78 @@ public void logUnused() {
         return objects;
     }
 
+    /**
+     * Instantiates given list of config providers and fetches the actual values of config variables from the config providers.
+     * returns a map of config key and resolved values.
+     * @param indirectVariables The map of config variables
+     * @param configProviders The map of config provider configs
+     * @param original The map of raw configs.
+     * @return map of resolved config variable.
+     */
+    @SuppressWarnings("unchecked")
+    private  Map<String, ?> resolveConfigVariables(Map<String, String> indirectVariables, Map<?, ?> configProviders, Map<String, Object> original) {
+        Map<String, ConfigProvider> providers;
+
+        if (configProviders == null || configProviders.isEmpty()) {
+            providers = instantiateConfigProvider(indirectVariables, original);
+        } else {
+
+            for (Map.Entry<?, ?> entry : configProviders.entrySet()) {
+                if (!(entry.getKey() instanceof String))
+                    configProviders.remove(entry.getKey());
```

Comment:
For non-string keys in originals, the code threw a ConfigException, why is this removing the entry silently?

Reply:
If we define configProviders as Map<String, ?> we wont need this check. Fixed it